test/spec/lifecycle-test.js

```diff
@@ -881,6 +881,100 @@ describe("Lifecycle tests", function () {
           });
         });
       });
+
+      describe("preventCollision=true and no compaction (#1933)", () => {
+        const resizeHandles = ["n", "e", "s", "w"];
+        const PreventCollisionContainer = ({ layoutA, layoutB }) => (
+          <ReactGridLayout
+            className="layout"
+            cols={12}
+            rowHeight={30}
+            width={1200}
+            preventCollision={true}
+            compactType={null}
+            resizeHandles={resizeHandles}
+          >
+            <div key="0" data-grid={layoutA}>
+              0
+            </div>
+            <div key="1" data-grid={layoutB}>
+              1
+            </div>
+          </ReactGridLayout>
+        );
+
+        it("Does not allow elements to move when resizing with no free space", () => {
+          const wrapper = mount(
+            <PreventCollisionContainer
+              layoutA={{ x: 0, y: 0, w: 1, h: 2, i: "0" }}
+              layoutB={{ x: 1, y: 0, w: 7, h: 2, i: "1" }}
+            />
+          );
+          const gridLayout = wrapper.find("ReactGridLayout");
+          const itemId = 0;
+          const gridItem0 = findGridItemByText(gridLayout, itemId);
+          const handleElement = findHandleForGridItem(gridItem0, "e");
+          const pos = getCurrentPosition(gridItem0);
+          const positionBeforeResize = getGridItemData(gridLayout, itemId);
+
+          // Resize right two columns
+          resizeTo(handleElement, true, pos, pos.left + colWidth * 2, pos.top);
+          const positionAfterResize = getGridItemData(gridLayout, itemId);
+          // Position shouldn't change because the system should preventCollisions
+          expect(positionAfterResize).toEqual(positionBeforeResize);
+        });
+
+        it("Allows elements to resize only within available free space", () => {
+          const wrapper = mount(
+            <PreventCollisionContainer
+              layoutA={{ x: 0, y: 0, w: 1, h: 2, i: "0" }}
+              layoutB={{ x: 2, y: 0, w: 7, h: 2, i: "1" }}
+            />
+          );
+          const gridLayout = wrapper.find("ReactGridLayout");
+          const itemId = 0;
+          const gridItem0 = findGridItemByText(gridLayout, itemId);
+          const handleElement = findHandleForGridItem(gridItem0, "e");
+          const pos = getCurrentPosition(gridItem0);
+          const positionBeforeResize = getGridItemData(gridLayout, itemId);
+
+          // Resize right two columns
+          resizeTo(handleElement, true, pos, pos.left + colWidth, pos.top);
+          resizeTo(handleElement, false, pos, pos.left + colWidth * 2, pos.top);
```

Comment:
This duplication of `resizeTo` calls here is intentional. We first move 1 column over (which is valid, no collisions), then we move another column over (which is invalid, there is collisions). This is to better simulate an actual drag.

The problem was with the collision detection and the logic that's in place is that if we went immediately from currentX/currentY -> newX/newY without any valid positions inbetween, the "last known good X/Y are the original position. This isn't realistic behaviour when a user actually drags their mouse, which is why I simulated it like this.

A better solution is probably to look into using `DragEvent` instead of `MouseEvent` for these simulations. I can take a look into that outside of this PR to have tests better simulate reality.